app/controllers/article_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class ArticleController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @articles = Article
+                .where(["user_id = ?", params[:id] || current_user.id])
```

Comment:
Why do you need an array here?
```suggestion
                .where("user_id = ?", params[:id] || current_user.id)
```